### ISSUE TITLE
SMRF latest version

### ISSUE THREAD

**Issue by scotthavens** [USDA-ARS-NWRC/awsm#68]
The latest version of SMRF breaks all sorts of things.

**Comment by jomey** [comment]
Is it time to think about a 'stable' SMRF API that libraries can integrate against, where SMRF is a dependency?

Feels like there were a couple of occasions recently that 'broke' things in AWSM and a more stable API could help here.

**Comment by scotthavens** [comment]
I think that we are getting close to have SMRF fairly stable again. These latest round of changes will really help with AWSM. Also, the main problem with AWSM is that there is a lot of repeated code from SMRF in AWSM, so a good cleanup of AWSM will help lessen the affects of SMRF changes. For example, the threading in AWSM is copy and pasted from SMRF.

Then once we have a better SMRF integration in AWSM, we can pin it to a specific minor version in the requirements, like `smrf>=0.11.0,<0.12.0` to ensure it stays with the minor version.

**Comment by jomey** [comment]
Got it. Agree that there should be little to no code duplication between the two modules.

**Comment by scotthavens** [comment]
Agreed, it's served us well and it's time to finally do some maintenance on it.
